### PR TITLE
refactor(cli): report progress when installing packages

### PR DESCRIPTION
**Pull request by thesayyn**
Closes SP-200 and SP-231.

### CODE REVIEW

**Comment by thesayyn** [comment]
@htuna07 first commit is irrelevant with the commit but I had to make those changes FYI.

File: stacks/api/function/pkgmanager/node/npm.ts [hunk 1]
```diff
@@ -1,29 +1,90 @@
...
+
+        if (chunk.indexOf("timing") != -1) {
+          progress += 1;
+        }
+
+        observer.next(Math.min(Math.round(progressAmountOfAStep * progress), 100));
+
+        stderr += chunk;
+      });
+      proc.on("time", () => console.log("DSADASDASDSAD"));
```
Comment (htuna07) [inline stacks/api/function/pkgmanager/node/npm.ts]: Probably you forgot to remove this line.

File: stacks/api/function/pkgmanager/test/node/npm.spec.ts [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+
+  it("should report progress", done => {
+    const progress = [];
+    npm
+      .install(cwd, "debug")
+      .pipe(distinctUntilChanged())
+      .subscribe({
+        next: p => progress.push(p),
+        complete: () => {
+          // prettier-ignore
```
Comment (htuna07) [inline stacks/api/function/pkgmanager/test/node/npm.spec.ts]: Why did we need to ignore this line? :)

Comment (thesayyn) [reply stacks/api/function/pkgmanager/test/node/npm.spec.ts]: Cause it is much more readable this way.

File: stacks/api/function/pkgmanager/node/npm.ts (hunk 1 above)
Comment (thesayyn) [reply stacks/api/function/pkgmanager/node/npm.ts]: Good catch. Thank you

**Pull request merged**